.github/workflows/test_helm_chart.yml

```diff
@@ -0,0 +1,155 @@
+      - main
+      - release-*
+    paths:
+      - "promxy-operator/**"
+      - "charts/**"
+      - "!**.md"
+
+env:
+  GO_VERSION: "1.23"
+
+jobs:
+  build:
+    concurrency:
+      group: build-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    name: Build and Unit Test
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.vars.outputs.version }}
+      pr: ${{ steps.pr.outputs.result }}
+    permissions:
+      packages: write
+    steps:
+      - name: Get PR ref
+        uses: actions/github-script@v7
+        id: pr
+        with:
+          script: |
+            const { data: pullRequest } = await github.rest.pulls.get({
+              ...context.repo,
+              pull_number: context.payload.pull_request.number,
+            });
+            return pullRequest
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{fromJSON(steps.pr.outputs.result).merge_commit_sha}}
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: false
+
+  setup-kind-cluster:
+    name: Setup Kind Cluster
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Install Kind
+        run: |
+          curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.20.0/kind-linux-amd64
+          chmod +x ./kind
+          sudo mv ./kind /usr/local/bin/kind
+      - name: Create Kind Cluster
+        run: |
+          kind create cluster --name kof-cluster
+
+    setup-local-registry:
+    name: Setup Local Docker Registry
+    runs-on: ubuntu-latest
+    needs: setup-kind-cluster
+    steps:
+      - name: Clone Docker Registry Repo
+        run: git clone https://github.com/distribution/distribution-library-image.git
+
+      - name: Build Docker Registry
+        run: |
+          cd distribution-library-image
+          docker build -t my-registry .
+          cd ..  # Move back to the previous directory
+
+      - name: Run Local Registry
+        run: |
+          docker run -d --restart=always -p "5000:5000" --name "custom-registry" my-registry
+
+      - name: Verify Registry is Running
+        run: |
+          echo "Checking registry availability..."
+          for i in {1..10}; do
+            if curl -s http://127.0.0.1:5000/v2/; then
+              echo "Registry is UP!"
+              exit 0
+            fi
+            echo "Waiting for registry..."
+            sleep 5
+          done
+          echo "Registry did not start!"
+          exit 1
+
+
+  setup-kcm:
+    name: Install and Apply KCM
+    runs-on: ubuntu-latest
+    needs: setup-kind-cluster
+    steps:
+      - name: Checkout KCM repository
+        uses: actions/checkout@v4
+        with:
+          repository: k0rdent/kcm  # ✅ Ensure we use k0rdent/kcm
+          ref: main  # Change if a different branch is needed
+          path: kcm-repo  # ✅ Check out into a separate directory
```

Comment:
Comments are not really needed as it's all documented in the action readme, please remove them.

Changing the branch for testing KOF against non-main KCM branch is useful, please add an input to select the branch
https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs